app/scripts/transaction-manager.js

```diff
@@ -0,0 +1,207 @@
+const EventEmitter = require('events')
+const extend = require('xtend')
+const TxProviderUtil = require('./lib/provider-utils')
+
+module.exports = class TransactionManager extends EventEmitter {
+  constructor (opts) {
+    super()
+    this.txList = opts.TxListFromStore || []
+    this._persistTxList = opts.setTxList
+    this._unconfTxCbs = {}
+    this.txLimit = opts.txLimit
+    this.provider = opts.provider
+  }
+
+//   Returns the tx list
+  getTxList () {
+    return this.txList
+  }
+
+  // Saves the new/updated txList.
+  // Function is intended only for internal use
+  _saveTxList (txList) {
+    this.txList = txList
+    this._persistTxList(txList)
+  }
+
+  // Adds a tx to the txlist
+  addTx (txData, onTxDoneCb) {
+    var txList = this.getTxList()
+    var txLimit = this.txLimit
+    if (txList.length > txLimit - 1) {
+      txList.shift()
+    }
+    txList.push(txData)
+    this._saveTxList(txList)
+    this.addOnTxDoneCb(txData.id, onTxDoneCb)
+    this.emit('unapproved', txData)
```

Comment:
we should consider instead or also emitting tx-specific events like `this.emit('unapproved:'+txMeta.id, txMeta)`